nucypher/cli/status.py

```diff
@@ -16,35 +16,85 @@
-from nucypher.cli.painting import paint_contract_status
+from nucypher.cli.painting import paint_contract_status, paint_stakers, paint_locked_tokens_status, paint_known_nodes
+from nucypher.cli.types import EIP55_CHECKSUM_ADDRESS
+from nucypher.config.characters import UrsulaConfiguration
 
 
 @click.command()
+@click.argument('action')
 @click.option('--poa', help="Inject POA middleware", is_flag=True, default=False)
 @click.option('--sync/--no-sync', default=False)
 @click.option('--geth', '-G', help="Run using the built-in geth node", is_flag=True)
 @click.option('--provider', 'provider_uri', help="Blockchain provider's URI", type=click.STRING, default="auto://")
+@click.option('--periods', help="Number of periods", type=click.INT, default=90)
+@click.option('--staking-address', help="Address of a NuCypher staker", type=EIP55_CHECKSUM_ADDRESS)
 @nucypher_click_config
-def status(click_config, provider_uri, sync, geth, poa):
+def status(click_config, action, provider_uri, sync, geth, poa, periods, staking_address):
     """
-    Echo a snapshot of live network metadata.
+    Echo a snapshot of live NuCypher Network metadata.
+
+    \b
+    Actions
+    -------------------------------------------------
+    view           Overall information of the NuCypher Network
+    stakers        Show relevant information about stakers
+    locked-tokens  Display a graph of evolution of locked tokens
+
     """
-    #
-    # Initialize
-    #
-    ursula_config = UrsulaConfiguration.from_configuration_file(filepath=config_file)
-    if not ursula_config.federated_only:
-        ursula_config.initialize_blockchain_interface()
-        ursula_config.acquire_agency()
-
-        # Contracts
-        paint_contract_status(click_config.emitter, ursula_config=ursula_config, click_config=click_config)
-
-    # Known Nodes
-    paint_known_nodes(emitter=click_config.emitter, ursula=ursula_config)
+
+    emitter = click_config.emitter
+    click.clear()
+    emitter.banner(NU_BANNER)
+
+    try:
+        ETH_NODE = None
+        if geth:
+            ETH_NODE = get_provider_process()
+
+        BlockchainInterfaceFactory.initialize_interface(provider_uri=provider_uri,
+                                                        provider_process=ETH_NODE,
+                                                        poa=poa)
+        blockchain = BlockchainInterfaceFactory.get_interface()
+        emitter.echo(message="Reading Latest Chaindata...")
+        blockchain.connect()
+    except Exception as e:
+        if click_config.debug:
+            raise
+        click.secho(str(e), bold=True, fg='red')
+        return  # Exit
+
+    # TODO: Allow --registry
+    registry = InMemoryContractRegistry.from_latest_publication()
+    staking_agent = ContractAgency.get_agent(StakingEscrowAgent, registry=registry)
+
+    if action == 'view':
+        ursula_config = UrsulaConfiguration.from_configuration_file(filepath=click_config.config_file)
```

Comment:
The status commands have become stateless and config agnostic *and* this is also a stale version of the original status.  A good way to fix both is to rebase this PR over #1157 .  I think it's OK to not display known nodes here, as a result.

Reply:
I made a rebased version for you to look at in `kprasch/stakers`.

Reply:
You rock @KPrasch! I'm using your rebased branch now.